### PR TITLE
Working fix for steam desktop authenticator and SteamAuth lib

### PR DESCRIPTION
**Pull request by jorgermduarte**
Hello everyone, this pull request is a fix since I noticed the main problem is on the SteamAuth lib.

I noticed a lot of scam attempts, and I decided to provide a full fix showing the code in the project pull request.

## Given the Steam Auth lib:

I followed the fix pull request from **Lyrapuff** since the endpoints from steamAuth as changed and right now the Steam Desktop Authenticator is not working. Given this problem, they will need to fix there first before making any changes to the Steam Desktop Authenticator itself.

This is the Lyrapuff pull request:

-> https://github.com/geel9/SteamAuth/pull/94

This is my commit on the fork with the extra fields, **_basically is the same as Lyrapuff but with some extra fields_**:

-> https://github.com/geel9/SteamAuth/commit/1dd1bae576bfc9d3ca24f2093df7426c405b2e41

## Given the Steam Desktop Authenticator

This pull request is based on the new implementation from **Lyrapuff** on the **SteamAuth** repository.

I created a commit and forked the project and added extra fields since I'm using the **Confirmation** class in the SDA too.

Basically right now, instead of using a browser inside the panel2 we are using a httpClient and render a panel with the trade list.

**_Everything is working properly and right now i have my staff working with my current SDA.**

## Notes

Anyone who know how to code, given this following fix, can build the SDA.exe in their machine literally copying 4/6 files, so that's why I will not provide a .exe here, doesn't make any sense.

Feel free to view the pull requests and give your opinions for this problem that will be probably fixed soon 😉  

# Attachment images of the implementation

image
image

### CODE REVIEW

**Comment by jorgermduarte** [comment]
**Reminder**: this is just a suggestion for a possible fix.

**Comment by chinap** [comment]
Hello, I can't add accounts to SDA because it failed to add a phone number because of: Error "GeneralFailure"

**Comment by jorgermduarte** [comment]
Hi, chinap, i will take a look at it later on and provide you some feedback, I only tested importing the maFile

**Comment by jorgermduarte** [comment]
**Update**: i changed the design of the trade list:

image

**Comment by allendaydayup** [comment]
> 

Can I download it from your warehouse and run it directly? I want to try it

**Comment by nicknike87** [comment]
@jorgermduarte 
I have build your project.
But  AutoConfirmMarketTransactions does not work, because all typical confirmations in market I have right now of the "unknow" type, not "MarketSellTransaction", and confirmations do not added to the collections "autoAcceptConfirmations" 

You can check your type of transactions in the market? 
Or is it just my problem with market.

**Comment by allendaydayup** [comment]
System.Reflection.TargetInvocationException
  HResult=0x80131604
  Message=调用的目标发生了异常。
  Source=mscorlib
  StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbacks()
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG& msg)
   at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.Run(Form mainForm)
   at Steam_Desktop_Authenticator.Program.Main(String[] args) in D:\gitee\steam-desktop-authenticator-jorgermduarte\Steam Desktop Authenticator\Program.cs:line 101

  此异常最初是在此调用堆栈中引发的: 
    Newtonsoft.Json.JsonConvert.DeserializeObject(string, System.Type, Newtonsoft.Json.JsonSerializerSettings)
    Newtonsoft.Json.JsonConvert.DeserializeObject<T>(string, Newtonsoft.Json.JsonSerializerSettings)
    Newtonsoft.Json.JsonConvert.DeserializeObject<T>(string)
    SteamAuth.SteamGuardAccount.FetchConfirmationInternal(string) (位于 SteamGuardAccount.cs 中)
    SteamAuth.SteamGuardAccount.FetchConfirmationsAsync() (位于 SteamGuardAccount.cs 中)
    System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
    System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
    System.Runtime.CompilerServices.TaskAwaiter<TResult>.GetResult()
    Steam_Desktop_Authenticator.MainForm.timerTradesPopup_Tick(object, System.EventArgs) (位于 MainForm.cs 中)
    System.Runtime.CompilerServices.AsyncMethodBuilderCore.ThrowAsync.AnonymousMethod__6_0(object)

内部异常 1:
ArgumentNullException: 值不能为 null。
Arg_ParamName_Name



When I turn on the automatic check confirmation message every 5 seconds, after a period of time, I will report the error above

**Comment by nikviktorovich** [comment]
It seems like the "ConfirmationsResponse" class is missing (file "ConfirmationFormWeb.cs", line 52: "var response = JsonConvert.DeserializeObject<ConfirmationsResponse>(httpResponse);"). Would you mind to check it?

**Comment by allendaydayup** [comment]
image

The interface '/mobileconf/getlist' occasionally returns null and should be evaluated within the program

**Comment by Jessecar96** [comment]
Thank you for the UI changes, I've integrated your changes into the updated version I put together today

**Comment by jorgermduarte** [comment]
No problem 😉 thank you
